### ISSUE TITLE
tkinter needed for versions >= 2.2 for <TropicalCyclone>.track_map() to work

### ISSUE THREAD

**Issue by ksgwxfan** [ksgwxfan/hurdat2parser#1]
`tkinter`-related methods were accessed in the code for `<TropicalCyclone>.track_map()` to assist in auto-resizing the `matplotlib` canvas. This affects `v2.2 <= versions <= 2.2.3` (current as of this post). But I do not have this listed as a dependency. I'll either add it as a dependency or work on making the code backend-independent (which I would rather).

I originally included it because I'm not satisfied with `matplotlib`s default window sizing but want it to accommodate the user's screen.

**Comment by lisalenorelowe** [comment]
Hi Kyle,

I'm working on a NASA Transform to Open Science project.  I am just starting to build it: [science-core, marine sciences](https://sciencecore-marine-sciences.github.io/intro.html), and part of it is best-track data.   I really like this package, it is clean and well documented, and doesn't have a zillion dependencies that can go wrong.

I'd like to use the package in Jupyter Notebook tutorials.  I installed tk and tried track_map and still got an error.  On searching, it looks like Tk is never expect to work in a Notebook.  I'm not a great Python programmer, but is there anything I can do to help you update track_map so it will use matplotlib and work in a Notebook?  I know you are not paid for this, and I appreciate your time.

Thanks,
Lisa

**Comment by ksgwxfan** [comment]
A hotfix was implemented in v2.2.3.1 that skips over tkinter-dependent lines.

**Comment by ksgwxfan** [comment]
As it was only 20 lines of code that were tkinter-dependent, I released a hotfix where I simply placed those lines within a generic try/except block so those lines are silently skipped upon an error. Try upgrading the hurdat2parser package (pip install --upgrade hurdat2parser) and then try.

Those lines I blocked out weren't vital to successful execution of the track_map method. Tkinter is one of the GUI backend options that the matplotlib package can use.
